src/platform_impl/apple/uikit/view.rs

```diff
@@ -512,4 +539,65 @@ impl WinitView {
         let mtm = MainThreadMarker::new().unwrap();
         app_state::handle_nonuser_events(mtm, touch_events);
     }
+
+    fn handle_insert_text(&self, text: &NSString) {
+        let window = self.window().unwrap();
+        let uiscreen = window.screen();
+        let window_id = RootWindowId(window.id());
+        unsafe {
+            let mtm = MainThreadMarker::new().unwrap();
+            // send individual events for each character
+            app_state::handle_nonuser_events(
+                mtm,
+                text.to_string().chars().map(|c| {
+                    EventWrapper::StaticEvent(Event::WindowEvent {
+                        window_id,
+                        event: WindowEvent::KeyboardInput {
+                            event: KeyEvent {
```

Comment:
I feel like this ought to be generating `Ime` events (new ones if need be) rather than `KeyEvent` events

Reply:
The Android IME (at least with the proposed patches) does generate the regular key events.

Some phone keyboards have an option to send things like `Control`, etc, so I'm not sure how stuff like that will go through regular `Commit` stuff though.

And there's probably nothing particularly bad in having input via regular keys, especially given that you can plug keyboard on some ios devices.

Reply:
Since UIKeyInput only generates basic key events (no autocomplete etc.) I think KeyEvents make more sense. I think a next step would be to implement UITextInput, which would allow for all the fancy IME stuff like autocomplete and autocorrect, then it would make sense to generate IME events

Reply:
Indeed, this is a good first step forwards, and we can always work on making IME better